### ISSUE TITLE
Property of struct called "shared" cause shader compile error

### ISSUE THREAD

**Issue by Dixeran** [servo/webrender#4227]
I was trying to run the basic example but with no luck. 

![image](https://user-images.githubusercontent.com/22679810/105631939-de6bbd80-5e8b-11eb-876d-25a9cd7adfb3.png)

```
ERROR 2021-01-24T12:31:54Z: webrender::device::gl: Failed to compile shader: cs_clip_rectangle
Vertex shader failed to compile with the following errors:
ERROR: 0:868: error(#132) Syntax error: "shared" parse error
ERROR: error(#273) 1 compilation errors.  No code generated
```

I'm using an AMD CPU with Radeon RX Vega 7 GPU and the latest driver:

```
Software Version - 2020.1204.0703.12704
Driver Version - 20.45.01.28-201204a-361901C-RadeonSoftwareAdrenalin2020
Provider - Advanced Micro Devices, Inc.
2D Driver Version - 8.1.1.1634
Direct3D® Version - 9.14.10.01471
OpenGL® Version - 26.20.11000.14757
AMD Audio Driver Version - 10.0.1.16
Vulkan™ Driver Version - 2.0.168
Vulkan™ API Version - 1.2.159
Windows Version - Windows 10 (64 bit)
```

If I switch to the standalone Nvidia graphic card, no error shows. So I tried to replace all "shared" in all shaders with another name, like this

```
struct ClipMaskInstanceRect {
    ClipMaskInstanceCommon shars;
    vec2 local_pos;
};
```

then there was no error reported. 
Firefox with webrender enabled didn't crash, so I'm not sure if this is a driver-related problem or if I missed something when running the example.

**Comment by kvark** [comment]
Filed https://bugzilla.mozilla.org/show_bug.cgi?id=1688983

**Comment by kvark** [comment]
Wrote a quick fix in https://phabricator.services.mozilla.com/D103103

**Comment by kvark** [comment]
fixed in d242f57b151b57067e015c7964f719a8f948a0ee

**Comment by Dixeran** [comment]
Confirm that this problem has been fixed.
